### PR TITLE
fix: location error

### PR DESCRIPTION
**Pull request by Mrkezii**
## Summary
* Add API endpoint to return single modite detail instead of relying on stale data coming from the KV store on cloudflare
* Update modites worker to remove the function to append modite profile because that data was stale and i could not tell what was updating it, plus it does not seem to be useful to the FE

## Full Gist

After running in circles for a while i realised that from the [modites](https://dash.cloudflare.com/1649832f2fb144695c75d5ae70be0e06/workers/view/modites) worker we were returning the `json` of all modites with their custom profile fields from slack (e.g location) but it turns out that data was stale because we were not getting the data from slack API directly but from the KV Store on cloudflare, i could not find what was responsible for updating the KV store but what i figured out was that slack required another API call to actually get the custom fields value. 

It turns out that we already had a [worker](https://dash.cloudflare.com/1649832f2fb144695c75d5ae70be0e06/workers/view/modite) for this but for some reason we were not using it on the FrontEnd. 

I proceeded to include that request to get the profile of a modite when we hit the individual URL and also remove the code that was appending the stale data from the modites worker file.

### CODE REVIEW

**Comment by mitchellsimoens** [comment]
<details><summary>Source Map Explorer Audits (0 failures)</summary>
<p>

## Source Map Explorer Audits

### /home/circleci/repo/build/static/js/main.004e5c95.chunk.js

| Label            | Value | Threshold | Success |
| ----             | :---: |   :---:   |  :---:  |
| index.tsx        | 677 B |   700 B   |    ✓    |
| serviceWorker.js | 316 B |  1.5 KB   |    ✓    |
| <unmapped>       | 147 B |   250 B   |    ✓    |

### /home/circleci/repo/build/static/js/page-modites.85cc96df.chunk.js

| Label                                                  | Value    | Threshold | Success |
| ----                                                   |  :---:   |   :---:   |  :---:  |
| components/GlobeModiteList/styles.module.scss          |  277 B   |           |    ✓    |
| components/NoResults/styles.module.scss                |  306 B   |           |    ✓    |
| components/GlobeModiteList/index.tsx                   | 1.12 KB  |           |    ✓    |
| hook/useBuffered.ts                                    |  239 B   |           |    ✓    |
| components/ModiteList/index.tsx                        |  562 B   |           |    ✓    |
| ... ents/NoResults/Rocket-Empty-State-Illustration.svg | 16.32 KB |           |    ✓    |
| components/NoResults/index.tsx                         |  435 B   |           |    ✓    |
| pages/Modites/index.tsx                                |  341 B   |   700 B   |    ✓    |
| ... e_modules/@babel/runtime/helpers/defineProperty.js |  147 B   |           |    ✓    |
| ... es/@babel/runtime/helpers/assertThisInitialized.js |  165 B   |           |    ✓    |
| components/ModiteImage/index.tsx                       |  529 B   |           |    ✓    |
| components/Time/index.tsx                              |  984 B   |           |    ✓    |
| ../node_modules/react-image/umd/index.js               | 3.38 KB  |           |    ✓    |
| ... ode_modules/@babel/runtime/helpers/objectSpread.js |  366 B   |           |    ✓    |
| ... /@babel/runtime/helpers/objectWithoutProperties.js |  280 B   |           |    ✓    |
| ... el/runtime/helpers/objectWithoutPropertiesLoose.js |  161 B   |           |    ✓    |
| ... e_modules/@babel/runtime/helpers/classCallCheck.js |  121 B   |           |    ✓    |
| ... node_modules/@babel/runtime/helpers/createClass.js |  252 B   |           |    ✓    |
| ... babel/runtime/helpers/possibleConstructorReturn.js |  119 B   |           |    ✓    |
| ../node_modules/@babel/runtime/helpers/typeof.js       |  504 B   |           |    ✓    |
| ... e_modules/@babel/runtime/helpers/getPrototypeOf.js |  172 B   |           |    ✓    |
| ../node_modules/@babel/runtime/helpers/inherits.js     |  268 B   |           |    ✓    |
| ... e_modules/@babel/runtime/helpers/setPrototypeOf.js |  135 B   |           |    ✓    |
| components/ModiteImage/styles.module.css               |  102 B   |           |    ✓    |
| components/ModiteImage/user-icon-placeholder.png       |   84 B   |           |    ✓    |
| components/Time/styles.module.css                      |  162 B   |           |    ✓    |
| components/ModiteList/Item.module.scss                 |  304 B   |           |    ✓    |
| components/ModiteList/Row.module.css                   |  172 B   |           |    ✓    |
| components/ModiteList/Item.tsx                         |  383 B   |           |    ✓    |
| components/ModiteList/Row.tsx                          |  265 B   |           |    ✓    |
| <unmapped>                                             |  132 B   |   150 B   |    ✓    |

</p>
</details>

<details><summary>Size Checks (0 failures)</summary>
<p>

## Size Checks

| Label                                                  | Value     | Threshold | Success |
| ----                                                   |   :---:   |   :---:   |  :---:  |
| build/static/js/main.004e5c95.chunk.js                 |   662 B   |   5 KB    |    ✓    |
| build/static/js/0.323541b4.chunk.js                    |  3.43 KB  |   1 MB    |    ✓    |
| build/static/js/10.dc2a7aea.chunk.js                   |  3.14 KB  |   1 MB    |    ✓    |
| build/static/js/11.4722c276.chunk.js                   |  5.36 KB  |   1 MB    |    ✓    |
| build/static/js/12.2a905dce.chunk.js                   |  3.27 KB  |   1 MB    |    ✓    |
| build/static/js/13.d95f0866.chunk.js                   |  2.19 KB  |   1 MB    |    ✓    |
| build/static/js/14.cfed0d28.chunk.js                   | 24.63 KB  |   1 MB    |    ✓    |
| build/static/js/15.9d8080be.chunk.js                   |  3.14 KB  |   1 MB    |    ✓    |
| build/static/js/16.7eb4a60f.chunk.js                   |  5.37 KB  |   1 MB    |    ✓    |
| build/static/js/17.20dbeba0.chunk.js                   |  3.14 KB  |   1 MB    |    ✓    |
| build/static/js/18.b432db5e.chunk.js                   |  5.36 KB  |   1 MB    |    ✓    |
| build/static/js/19.bd9d0224.chunk.js                   |  2.2 KB   |   1 MB    |    ✓    |
| build/static/js/7.75ed7e08.chunk.js                    | 237.96 KB |   1 MB    |    ✓    |
| build/static/js/8.0eb019fc.chunk.js                    |  9.09 KB  |   1 MB    |    ✓    |
| build/static/js/9.ea7ed59e.chunk.js                    | 36.61 KB  |   1 MB    |    ✓    |
| ... ild/static/js/page-modite-detail.7f978879.chunk.js |  4.31 KB  |   1 MB    |    ✓    |
| build/static/js/page-modites.85cc96df.chunk.js         |  8.06 KB  |   1 MB    |    ✓    |
| ... ld/static/js/page-project-detail.3aee3513.chunk.js |  3.63 KB  |   1 MB    |    ✓    |
| build/static/js/page-projects.e4906040.chunk.js        |  1016 B   |   1 MB    |    ✓    |
| build/static/js/runtime~main.d6e4a78f.js               |  1.76 KB  |   10 KB   |    ✓    |
| build/static/media/logo-android.90adbe1d.svg           |   435 B   |   3 KB    |    ✓    |
| build/static/media/logo-angular.c42a69d0.svg           |   201 B   |   3 KB    |    ✓    |
| build/static/media/logo-apple.b2c62065.svg             |   334 B   |   3 KB    |    ✓    |
| build/static/media/logo-bitbucket.4f9455bc.svg         |   243 B   |   3 KB    |    ✓    |
| build/static/media/logo-bitcoin.4fd2e1b8.svg           |   538 B   |   3 KB    |    ✓    |
| build/static/media/logo-buffer.c60ed6c2.svg            |   333 B   |   3 KB    |    ✓    |
| build/static/media/logo-chrome.f2282752.svg            |   761 B   |   3 KB    |    ✓    |
| ... d/static/media/logo-closed-captioning.5a1ebda9.svg |   511 B   |   3 KB    |    ✓    |
| build/static/media/logo-codepen.5bd8e715.svg           |   876 B   |   3 KB    |    ✓    |
| build/static/media/logo-css3.5c44e01d.svg              |   260 B   |   3 KB    |    ✓    |
| build/static/media/logo-designernews.948250b1.svg      |   259 B   |   3 KB    |    ✓    |
| build/static/media/logo-dribbble.d4f5af0d.svg          |   562 B   |   3 KB    |    ✓    |
| build/static/media/logo-dropbox.3f0350cc.svg           |   229 B   |   3 KB    |    ✓    |
| build/static/media/logo-euro.a064e736.svg              |   326 B   |   3 KB    |    ✓    |
| build/static/media/logo-facebook.46c0bd59.svg          |   249 B   |   3 KB    |    ✓    |
| build/static/media/logo-flickr.52aaffb8.svg            |   197 B   |   3 KB    |    ✓    |
| build/static/media/logo-foursquare.da8be88c.svg        |   533 B   |   3 KB    |    ✓    |
| build/static/media/logo-freebsd-devil.08a0c13b.svg     |   650 B   |   3 KB    |    ✓    |
| ... d/static/media/logo-game-controller-a.12501789.svg |   412 B   |   3 KB    |    ✓    |
| ... d/static/media/logo-game-controller-b.17a42e12.svg |   462 B   |   3 KB    |    ✓    |
| build/static/media/logo-github.8a1ba32d.svg            |   469 B   |   3 KB    |    ✓    |
| build/static/media/logo-google.4af031c1.svg            |   321 B   |   3 KB    |    ✓    |
| build/static/media/logo-googleplus.f5e50927.svg        |   350 B   |   3 KB    |    ✓    |
| build/static/media/logo-hackernews.c92a375d.svg        |   156 B   |   3 KB    |    ✓    |
| build/static/media/logo-html5.0263f883.svg             |   252 B   |   3 KB    |    ✓    |
| build/static/media/logo-instagram.c1dbcbd5.svg         |   334 B   |   3 KB    |    ✓    |
| build/static/media/logo-ionic.b69346c0.svg             |   341 B   |   3 KB    |    ✓    |
| build/static/media/logo-ionitron.057a6c2f.svg          |   883 B   |   3 KB    |    ✓    |
| build/static/media/logo-javascript.bbbb150f.svg        |   450 B   |   3 KB    |    ✓    |
| build/static/media/logo-linkedin.8dc04bcd.svg          |   324 B   |   3 KB    |    ✓    |
| build/static/media/logo-markdown.ae7fedcb.svg          |   283 B   |   3 KB    |    ✓    |
| build/static/media/logo-model-s.7d58a556.svg           |  1.03 KB  |   3 KB    |    ✓    |
| build/static/media/logo-no-smoking.e3c2c91b.svg        |   811 B   |   3 KB    |    ✓    |
| build/static/media/logo-nodejs.5d51536d.svg            |   902 B   |   3 KB    |    ✓    |
| build/static/media/logo-npm.d631bf07.svg               |   217 B   |   3 KB    |    ✓    |
| build/static/media/logo-octocat.3bdb0ef9.svg           |   689 B   |   3 KB    |    ✓    |
| build/static/media/logo-pinterest.870c40e9.svg         |   434 B   |   3 KB    |    ✓    |
| build/static/media/logo-playstation.80ff1244.svg       |   687 B   |   3 KB    |    ✓    |
| build/static/media/logo-polymer.7d0fc352.svg           |   167 B   |   3 KB    |    ✓    |
| build/static/media/logo-python.7759b8cd.svg            |   747 B   |   3 KB    |    ✓    |
| build/static/media/logo-reddit.4ee9a163.svg            |   762 B   |   3 KB    |    ✓    |
| build/static/media/logo-rss.b2f31e4d.svg               |   242 B   |   3 KB    |    ✓    |
| build/static/media/logo-sass.9796e80b.svg              |  1.64 KB  |   3 KB    |    ✓    |
| build/static/media/logo-skype.2ff8f84c.svg             |   766 B   |   3 KB    |    ✓    |
| build/static/media/logo-slack.e88bf77b.svg             |   362 B   |   3 KB    |    ✓    |
| build/static/media/logo-snapchat.9c8aede7.svg          |  1.15 KB  |   3 KB    |    ✓    |
| build/static/media/logo-steam.81116cd2.svg             |   529 B   |   3 KB    |    ✓    |
| build/static/media/logo-tumblr.a3f62e33.svg            |   314 B   |   3 KB    |    ✓    |
| build/static/media/logo-tux.edfd5daf.svg               |  1.26 KB  |   3 KB    |    ✓    |
| build/static/media/logo-twitch.cd5a611b.svg            |   187 B   |   3 KB    |    ✓    |
| build/static/media/logo-twitter.3ee3ca38.svg           |   384 B   |   3 KB    |    ✓    |
| build/static/media/logo-usd.8b88ec03.svg               |  1006 B   |   3 KB    |    ✓    |
| build/static/media/logo-vimeo.8f0c9d7b.svg             |   450 B   |   3 KB    |    ✓    |
| build/static/media/logo-vk.5d4d40d5.svg                |   578 B   |   3 KB    |    ✓    |
| build/static/media/logo-whatsapp.9bb37426.svg          |   626 B   |   3 KB    |    ✓    |
| build/static/media/logo-windows.45fff7fb.svg           |   170 B   |   3 KB    |    ✓    |
| build/static/media/logo-wordpress.f033eb37.svg         |   615 B   |   3 KB    |    ✓    |
| build/static/media/logo-xbox.d065e077.svg              |   437 B   |   3 KB    |    ✓    |
| build/static/media/logo-xing.ba7d9b79.svg              |   241 B   |   3 KB    |    ✓    |
| build/static/media/logo-yahoo.1738d03e.svg             |   205 B   |   3 KB    |    ✓    |
| build/static/media/logo-yen.94a9e893.svg               |   186 B   |   3 KB    |    ✓    |
| build/static/media/logo-youtube.bad4875e.svg           |   293 B   |   3 KB    |    ✓    |
| build/favicon.ico                                      | 10.87 KB  |   11 KB   |    ✓    |
| build/index.html                                       |  3.01 KB  |   5 KB    |    ✓    |
| build/manifest.json                                    |   244 B   |   500 B   |    ✓    |
| ... cache-manifest.3bdc9eca1a9fa4208d3d9c7de75b3085.js |   177 B   |   500 B   |    ✓    |
| build/service-worker.js                                |   905 B   |  1.2 KB   |    ✓    |
| build                                                  |  6.81 MB  |   20 MB   |    ✓    |

</p>
</details>

<details><summary>Heap Snapshot Checks (0 failures)</summary>
<p>

## Heap Snapshot Checks

| Label            | Value   | Threshold | Success |
| ----             |  :---:  |   :---:   |  :---:  |
| Documents        |    6    |    11     |    ✓    |
| Frames           |    2    |     5     |    ✓    |
| JSHeapTotalSize  | 9043968 |           |    ✓    |
| JSHeapUsedSize   | 7308944 |           |    ✓    |
| LayoutCount      |    4    |     5     |    ✓    |
| Nodes            |   114   |    800    |    ✓    |
| RecalcStyleCount |    7    |    10     |    ✓    |

</p>
</details>

<details><summary>Axe Audits (0 failures)</summary>
<p>

## Axe Audits

| Label                             | Value    | Threshold | Success |
| ----                              |  :---:   |   :---:   |  :---:  |
| aria-hidden-body                  |          |   none    |    ✓    |
| aria-hidden-focus                 |          |   none    |    ✓    |
| avoid-inline-spacing              |          |   none    |    ✓    |
| color-contrast                    |          |   none    |    ✓    |
| document-title                    |          |   none    |    ✓    |
| duplicate-id                      |          |   none    |    ✓    |
| html-has-lang                     |          |   none    |    ✓    |
| html-lang-valid                   |          |   none    |    ✓    |
| landmark-no-duplicate-banner      |          |   none    |    ✓    |
| landmark-no-duplicate-contentinfo |          |   none    |    ✓    |
| meta-viewport-large               |          |   none    |    ✓    |
| meta-viewport                     |          |   none    |    ✓    |
| landmark-one-main                 | moderate | moderate  |    ✓    |
| page-has-heading-one              | moderate | moderate  |    ✓    |
| region                            | moderate | moderate  |    ✓    |

</p>
</details>

<details><summary>Lighthouse Audits (0 failures)</summary>
<p>

## Lighthouse Audits

| Label               | Value | Threshold | Success |
| ----                | :---: |   :---:   |  :---:  |
| Performance         |  98   |    60     |    ✓    |
| Accessibility       |  100  |    100    |    ✓    |
| Best Practices      |  100  |    90     |    ✓    |
| SEO                 |  100  |    100    |    ✓    |
| Progressive Web App |  72   |    70     |    ✓    |

</p>
</details>

**Review by grgur — APPROVED** [review]
This looks great @Mrkezii! Thank you for all this effort.

Could you just double check that the code is formatted with prettier? I thought I saw a couple of alignment differences, that's all.

**Comment by Mrkezii** [comment]
Prettier format seems good from my end @grgur, i had added a small change to the `lint-staged` too

**Comment by grgur** [comment]
There seems to be a TS error that surfaced

```sh
#!/bin/bash -eo pipefail
yarn check:compile
yarn run v1.22.5
$ tsc --noEmit
src/components/App/index.tsx:184:10 - error TS2740: Type '{ mode: "md"; debounce: number; placeholder: string; value: string | undefined; onIonChange: (event: CustomEvent<SearchbarChangeEventDetail>) => void; className: string; }' is missing the following properties from type 'Pick<IonSearchbar & ReactProps & { ref?: RefObject<HTMLIonSearchbarElement> | undefined; children?: ReactNode; }, "dir" | "slot" | "style" | "title" | "className" | "hidden" | ... 180 more ... | "showCancelButton">': dir, slot, style, title, and 158 more.

184         <IonSearchbar
             ~~~~~~~~~~~~


Found 1 error.

error Command failed with exit code 1.
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.

Exited with code exit status 1
CircleCI received exit code 1
```

**Comment by Mrkezii** [comment]
@grgur 🤔 

That's a weird one, turns out we have the same ts error on the existing master branch. I'll take a look at it.

File: public/mock/modites/user.json [hunk 1]
```diff
@@ -1,50 +1,58 @@
...
-    "phone": "+1 208 555-1212",
-    "skype": "userA",
-    "real_name": "Finn Human",
-    "real_name_normalized": "Finn Human",
-    "display_name": "Finn",
-    "display_name_normalized": "Finn",
+    "title": "Doing whatever it takes @ Modus since 2012",
+    "phone": "+1 571 236 0938",
+    "skype": "rbulling",
+    "real_name": "Richard Bullington-McGuire",
```
Comment (obscurerichard) [inline public/mock/modites/user.json:7]: I'm honored to be the mock data subject :)

🥳

**Comment by Mrkezii** [comment]
@grgur this issue we face seems to come from either Ionic or Typescript but I cannot seem to figure out why it's an error because even when I follow the exact thing indicated in the [docs](https://ionicframework.com/docs/api/searchbar), I still get the error.

My suggestion is to take away that check from the CI process for now till i find a way for that to be fixed.

**Comment by grgur** [comment]
I'll merge so this doesn't become a blocker. Thank you for working on it @Mrkezii

**Pull request merged**